opslevel/resource_opslevel_integration_azure_resources.go

```diff
@@ -0,0 +1,223 @@
+package opslevel
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/opslevel/opslevel-go/v2024"
+)
+
+var _ resource.ResourceWithConfigure = &IntegrationAzureResourcesResource{}
+
+var _ resource.ResourceWithImportState = &IntegrationAzureResourcesResource{}
+
+func NewIntegrationAzureResourcesResource() resource.Resource {
+	return &IntegrationAzureResourcesResource{}
+}
+
+type IntegrationAzureResourcesResource struct {
+	CommonResourceClient
+}
+
+type IntegrationAzureResourcesResourceModel struct {
+	Aliases        types.List   `tfsdk:"aliases"`
+	ClientId       types.String `tfsdk:"client_id"`
+	ClientSecret   types.String `tfsdk:"client_secret"`
+	Id             types.String `tfsdk:"id"`
+	LastSyncedAt   types.String `tfsdk:"last_synced_at"`
+	Name           types.String `tfsdk:"name"`
+	SubscriptionId types.String `tfsdk:"subscription_id"`
+	TenantId       types.String `tfsdk:"tenant_id"`
+}
+
+func NewIntegrationAzureResourcesResourceModel(ctx context.Context, planModel IntegrationAzureResourcesResourceModel, azureResourcesIntegration opslevel.Integration) (IntegrationAzureResourcesResourceModel, diag.Diagnostics) {
```

Comment:
I think we've decided to call this `givenModel` instead of `planModel` given that sometimes its the stateModel and sometimes its the planModel

```suggestion
func NewIntegrationAzureResourcesResourceModel(ctx context.Context, givenModel IntegrationAzureResourcesResourceModel, azureResourcesIntegration opslevel.Integration) (IntegrationAzureResourcesResourceModel, diag.Diagnostics) {
```